src/Carter/ICarterModule.cs

```diff
@@ -64,6 +66,19 @@ protected CarterModule(string basePath)
     public CarterModule RequireAuthorization()
     {
         this.requiresAuthorization = true;
+        this.authorizationPolicyNames = null;
+        return this;
+    }
+
+    /// <summary>
+    /// Add authorization to all routes
+    /// </summary>
+    /// <param name="policyNames">A collection of policy names. If empty, the default authorization policy will be used.</param>
+    /// <returns></returns>
+    public CarterModule RequireAuthorization(params string[] policyNames)
+    {
+        this.requiresAuthorization = true;
```

Comment:
I'm not checking whether `policyNames` is `null` here.

If it is, then the module will require the default authorization policy.
I suppose it could lead to unexpected behaviors for consumers.

Let me know if you think we should add a null check.

Reply:
I think it's fine as the underlying stuff will blow if they supply null

Reply:
That's the thing, I don't think it will as we'll call the parameter-less overload of `RequireAuthorization` if they do.

Reply:
Actually because it's params they could just call RequiresAuthorization() with no string values so you could get rid of the method that doesn't have any args

Reply:
You'd have to change this: https://github.com/CarterCommunity/Carter/pull/310/files#diff-ff4fe2902bfc85b0e2bba954c7b04ac67fb33dbd0c34ea6319c9ae7182dba825R44 to check for an empty array instead of null too, but that's a nicer way of doing it imo

Reply:
nope, no need i don't think as we just pass it to aspnet core. if our array is populated great, if it's empty great but we just call `group.RequireAuthorization(carterModule.authorizationPolicyNames);`

Reply:
Yep, OK.

Reply:
Done in https://github.com/CarterCommunity/Carter/pull/310/commits/b0c3100ca66bd8eeaee4e28ca48ffbc5255d8395.